ext/bg/js/backend.js

```diff
@@ -565,23 +597,48 @@ class Backend {
     // Command handlers
 
     async _onCommandSearch(params) {
-        const url = chrome.runtime.getURL('/bg/search.html');
-        if (!(params && params.newTab)) {
-            try {
-                const tab = await Backend._findTab(1000, (url2) => (
-                    url2 !== null &&
-                    url2.startsWith(url) &&
-                    (url2.length === url.length || url2[url.length] === '?' || url2[url.length] === '#')
-                ));
-                if (tab !== null) {
-                    await Backend._focusTab(tab);
-                    return;
+        const {mode, query} = params || {mode: 'sameTab'};
+
+        const options = await this.getOptions(this.optionsContext);
+        const {popupWidth, popupHeight} = options.general;
+
+        const baseUrl = chrome.runtime.getURL('/bg/search.html');
+        const queryParams = {mode};
+        if (query && query.length > 0) { queryParams.query = query; }
+        const queryString = new URLSearchParams(queryParams).toString();
+        const url = `${baseUrl}?${queryString}`;
+
+        switch (mode) {
+            case 'sameTab':
+                try {
+                    const tab = await Backend._findTab(1000, (url2) => (
+                        url2 !== null &&
+                        url2.startsWith(url) &&
+                        (url2.length === url.length || url2[url.length] === '?' || url2[url.length] === '#')
+                    ));
+                    if (tab !== null) {
+                        await Backend._focusTab(tab);
+                        return;
+                    }
+                } catch (e) {
+                    // NOP
                 }
-            } catch (e) {
-                // NOP
-            }
+                chrome.tabs.create({url});
+                return;
+            case 'newTab':
+                chrome.tabs.create({url});
+                return;
+            case 'popup':
+                if (this.popupWindow !== null) {
```

Comment:
Firefox for Android doesn't support `chrome.windows`. While this feature shouldn't be enabled on Android, validate `chrome.windows` is valid before using.

https://github.com/FooSoft/yomichan/blob/d5708de4eed15567e14e0c1fd4998561eee1680e/ext/bg/js/backend.js#L705-L708